custom_components/pun_sensor/sensor.py

```diff
@@ -113,88 +112,88 @@ def __init__(self, coordinator: PUNDataUpdateCoordinator, tipo: int) -> None:
         self._native_value = 0
 
     def _handle_coordinator_update(self) -> None:
-        """Gestisce l'aggiornamento dei dati dal coordinator"""
-        self._available = self.coordinator.orari[self.tipo] > 0
-        if (self._available): self._native_value = self.coordinator.pun[self.tipo]
+        """Gestisce l'aggiornamento dei dati dal coordinator."""
+        if len(self.coordinator.pun_data.pun[self.fascia]) > 0:
+            self._available = True
+            self._native_value = self.coordinator.pun_values.value[self.fascia]
+        # special case for F23 because we don't have them in the dict of values
+        # can we compare against calculated value? if it's not 0 then it's available?
+        if (
+            self.fascia == Fascia.F23
+            and self.coordinator.pun_values.value[self.fascia] != 0
+        ):
+            self._available = True
+            self._native_value = self.coordinator.pun_values.value[self.fascia]
```

Comment:
Questa parte non sono sicuro di averla capita. `Fascia.F23` non è anch'essa presente nella classe `PunData` qui?
https://github.com/moddroid94/pun_sensor/blob/32e280d2bf00774354e58aff434169e407195e22/custom_components/pun_sensor/interfaces.py#L11

Ad occhio mi pare che questo ramo venga eseguito sempre.

Reply:
Eh in realta' credo di aver lasciato quell'F23 per "sbaglio" perche' nel file xml non c'e' la lista di valori per F23, lo avevo messo pensando di riempirlo ma non conviene allocare memoria senza senso, come prima calcola solo il valore finale, quindi nel caso in cui chiede la fascia F23 devo fare un check che il valore calcolato da F2 e F3 non sia 0, altrimenti significa che mancano dati